elements/lisk-dpos/src/dpos.ts

```diff
@@ -208,4 +209,34 @@ export class Dpos {
 			delegateListRoundOffset,
 		});
 	}
+
+	// This function is used in block_processor_v2 to check the dpos compliance and update/validate the reward
+	// tslint:disable-next-line: prefer-function-over-method
+	public async isDPoSProtocolCompliant(
+		blockHeader: BlockHeader,
+		store: StateStore,
+	): Promise<boolean> {
+		const delegateForgedBlocks = store.consensus.lastBlockHeaders.filter(
```

Comment:
this line needs to reflect
>  in the previous round or in the same round of B

from https://github.com/LiskHQ/lips/blob/master/proposals/lip-0022.md#validating-new-block-header-property